controllers/cloud.redhat.com/pool_controller.go

```diff
@@ -136,6 +136,16 @@ func (r *NamespacePoolReconciler) handleErrorNamespaces(ctx context.Context, err
 			return fmt.Errorf("handleErrorNamespace error: Couldn't delete namespace: %v", err)
 		}
 
+		err = DeleteSubscriptionPrometheusOperator(ctx, r.Client, nsName)
```

Comment:
We have this (mostly) same block three times. I wonder if we could extract this out to another function that's the single source.